--- ../rakudo/tools/build/gen-version.pl
+++ ../rakudo/tools/build/gen-version.pl
@@ -28,7 +28,7 @@ my $builddate = strftime('%Y-%m-%dT%H:%M:%SZ', gmtime);
 
 print <<"END_VERSION";
 sub hll-config(\$config) {
-    \$config<implementation>   := 'Rakudo';
+    \$config<implementation>   := 'Rakudo Star';
     \$config<version>          := '$version';
     \$config<release-number>   := '$release';
     \$config<codename>         := '$codename';
